### PR TITLE
squash! Discovery Tools for OCI Image

### PR DESCRIPTION
**Pull request by wking**
Pull Go host-based image name parsing out into its own package and add tests.

A fixup with part of my suggestions for #16.

### CODE REVIEW

File: tools/hostbasedimagenames/main.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
+package hostbasedimagenames
```
Comment (xiekeyang) [inline tools/hostbasedimagenames/main.go]: Could this name of `main.go` be changed? I feel it is improper to name a package file as `main.go`

File: tools/hostbasedimagenames/main.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+import (
+	"fmt"
+	"regexp"
+)
+
+var (
+	// TODO: This does not conform to https://tools.ietf.org/html/rfc3986 yet.
```
Comment (xiekeyang) [inline tools/hostbasedimagenames/main.go]: The comment change as same to [1]

[1]: https://github.com/xiekeyang/oci-discovery/blob/master/oci_discovery/host_based_image_names/__init__.py#L18-L21

**Comment by xiekeyang** [comment]
And one more question: after this landed, `go` and `python` will coexist in this project. Should we keep improving only `go`, and remove `python`?  Maintaining two languages seems to be redundant.

File: tools/hostbasedimagenames/main.go (hunk 2 above)
Comment (wking) [reply tools/hostbasedimagenames/main.go]: > The comment change as same to [1][]

Addressed with e3e4231 → 3bb506e, which updated the regexp to be complete (analogous to #18).

[1]: https://github.com/xiekeyang/oci-discovery/blob/master/oci_discovery/host_based_image_names/__init__.py#L18-L21

File: tools/hostbasedimagenames/main.go (hunk 1 above)
Comment (wking) [reply tools/hostbasedimagenames/main.go]: > Could this name of `main.go` be changed?

I've renamed it to `hostbasedimagenames.go` with e3e4231 → 3bb506e.

**Comment by xiekeyang** [comment]
LGTM

Some topics is still under discussion, we can patch them later.

**Comment by wking** [comment]
> Should we keep improving only go, and remove python? Maintaining two languages seems to be redundant.

I don't mind keeping both in parallel during the mockup stage.  If/when this project moves under the OCI, I think we want a spec-only repo with the `*.md` that is distribution-specific (everything but `cas-engine-protocols.md` and `cas-template.md`, which I think belong in image-spec).  Then we can have `go-distribution` with the Go implementation and `py-distribution` (or whatever) with the Python implemenation.  I don't mind if a Python repository ends up under the OCI or not; and I don't mind when it gets dropped from this repo.  If you'd rather drop it now, I can create my own `wking/oci-distribution` repository now.

**Comment by xiekeyang** [comment]
>  If you'd rather drop it now, I can create my own `wking/oci-distribution` repository now.

I won't, keeping both is OK.

**Pull request merged**